### ISSUE TITLE
Find and fix memory leaks

### ISSUE THREAD

**Issue by justinchuby** [onnx/onnx#4939]
Enable ASan and fix memory leaks following #4823 

@jcwchen @gramalingam

**Comment by justinchuby** [comment]
- https://github.com/onnx/onnx/issues/4287

**Comment by zhenhuaw-me** [comment]
Hi @justinchuby, if there is instruction about how to enable ASan/LSan in ONNX and run?

We recently see more segfault reports of tools that are built on ONNX package. They usually stuck or crash in protobuf's `CopyFrom()` function which is invoked by `onnx.save()`. And the system memory (hundreds of GB) are all taken by the process.

cc @kevinch-nv

**Comment by justinchuby** [comment]
I don’t think we have an option yet. https://github.com/onnx/onnx/blob/main/CMakeLists.txt is what we will have to edit. @jcwchen is that right?

Contribution/help would be much appreciated!

**Comment by justinchuby** [comment]
This is a previously failed attempt: https://github.com/onnx/onnx/pull/5053
